frontend/components/CourseActions/components/StatsModal/index.js

```diff
@@ -0,0 +1,92 @@
+import TogglerAndModal from '~/components/TogglerAndModal';
+import css from './index.scss';
+import { Link }        from 'react-router-dom';
+import api from '~/api';
+
+class StatsModal extends React.Component {
+  static propTypes = {
+    toggler: PropTypes.element.isRequired,
+    course: PropTypes.object.isRequired,
+    learners: PropTypes.array.isRequired,
+    currentUser: PropTypes.object.isRequired,
+    author: PropTypes.object.isRequired
+  }
+
+  state = {
+    learners: this.props.learners,
+    stats: [],
+    speUpdate: {}
+  }
+
+  getStats = () =>
+    api.CourseApi.getStudentsStats(
+      (spe) => spe.status === 'success',
+      { courseId: this.props.course.id, authorId: this.props.author.id }
+    )
+    .then((payload) => {
+      this.state.stats = []
```

Comment:
In react we should do `this.setState({ stats: [] })` instead of assigning to `this.state` directly, because only `this.setState` updates the UI automatically (and makes sure only the updates parts of the UI are refreshed).

Reply:
Got it, I've changed that too in the 3rd commit. Once I'm done with the exam session I'll check the React documentation to get a better feeling of how things go.